flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -1680,15 +1680,11 @@ class ScalarExprLowering {
     // Explicitly map statement function host associated symbols to their
     // parent scope lowered symbol box.
     for (const Fortran::semantics::SymbolRef &sym :
-         Fortran::evaluate::CollectSymbols(*details.stmtFunction())) {
+         Fortran::evaluate::CollectSymbols(*details.stmtFunction()))
       if (const auto *details =
-              sym->detailsIf<Fortran::semantics::HostAssocDetails>()) {
-        if (!symMap.lookupSymbol(*sym)) {
-          symMap.addSymbol(
-              *sym, symMap.lookupSymbol(details->symbol()).toExtendedValue());
-        }
-      }
-    }
+              sym->detailsIf<Fortran::semantics::HostAssocDetails>())
+        if (!symMap.lookupSymbol(*sym))
+          symMap.addSymbol(*sym, gen(details->symbol()));
```

Comment:
Will this call `gen` twice if the same symbol occurs multiple times in the statement function? And is that OK?

Reply:
No it won't because the `lookupSymbol` will prevent that after the symbol is added the first time to the symbol map. Calling `gen` on a symbol is actually a simple lookup, it does not produce any IR, so it's fine in general to call it multiple times.